src/main/java/duke/DataManager.java

```diff
@@ -0,0 +1,156 @@
+
+        for (String task : loadedTasksStringList) {
+
+            TaskType taskType;
+
+            // Get task type while catching unknown-task-type error
+            try {
+                taskType = getTaskType(task.charAt(TASK_TYPE_INDICATOR));
+            } catch (DukeException e) {
+                // Print error message to user and terminate loading of data file if task type is not found
+                DisplayManager.printDukeErrorMessage(e.getMessage());
+                return;
+            }
+
+            // Using enums to check the type of task we are dealing with.
+            // Unlikely to go to the default statement since enums are fixed values
+            switch(taskType) {
+            case TODO:
+                processTodoTask(task, tasks);
+                break;
+            case DEADLINE:
+                processDeadlineTask(task, tasks);
+                break;
+            case EVENT:
+                processEventTask(task, tasks);
+                break;
+            default:
+                return;
+            }
+        }
+    }
+
+    // If folder (data) and file (duke.txt) for saving and loading does not exist yet, create one for user.
+    private static void createSaveLocation() throws IOException {
+        // Create 'data' folder if it does not already exists
+        if (!Files.exists(DATA_DIRECTORY)) {
+            Files.createDirectory(DATA_DIRECTORY);
+        }
+        // Create 'duke.txt' file if it does not already exists
+        if (!Files.exists(DATA_PATH)) {
+            Files.createFile(DATA_PATH);
+        }
+    }
+
+    // Identifies the given task type
+    private static TaskType getTaskType(char taskTypeIndicator) throws DukeException {
+        // No breaks required as return statements are used
+        switch(taskTypeIndicator) {
+        case 'T':
+            return TaskType.TODO;
+        case 'D':
+            return TaskType.DEADLINE;
+        case 'E':
+            return TaskType.EVENT;
+        default:
+            throw new DukeException(ErrorTypeManager.ERROR_UNKNOWN_TASK_INDICATOR);
+        }
+    }
+
+    // Identify if task was marked as done before and mark it as done in current session if yes
+    private static void markTaskDoneIfDone (String task, ArrayList<Task> tasks) {
+        if (task.contains(Task.TICK_SYMBOL)) {
+            tasks.get(Task.getTaskCount() - 1).markAsDone();
+        }
+    }
+
+    // Exceptions are already handled in Duke.java and hence we will assume these are appropriate tasks
+    // No need to notify user that the task has been added too
+    private static void processTodoTask(String task, ArrayList<Task> tasks) {
+        String taskDescription = task.substring(START_OF_TASK_DESCRIPTION);
+
+        // Load saved to_do task into current tasks
+        tasks.add(new Todo(taskDescription));
+
+        // Recover its isDone status
+        markTaskDoneIfDone(task, tasks);
```

Comment:
The method name `markTaskDoneIfDone` might be confusing since there exists a `markAsDone`, to improve readability of code perhaps you could consider renaming it?